packages/client/src/LocClient.ts

```diff
@@ -0,0 +1,332 @@
+import {
+    File as BlockchainFile,
+    MetadataItem,
+    Link,
+    CollectionItem,
+    LocType,
+    LegalOfficerCase,
+    VoidInfo,
+} from '@logion/node-api/dist/Types';
+import { UserIdentity, LegalOfficer } from "./Types";
+import { NetworkState } from "./NetworkState";
+import { LegalOfficerEndpoint } from "./SharedClient";
+import { AxiosFactory } from "./AxiosFactory";
+import { LogionNodeApi, UUID } from "node-api/dist";
+import { getLegalOfficerCase, addCollectionItem, getCollectionItem } from "@logion/node-api/dist/LogionLoc";
+import { AxiosInstance } from 'axios';
+import { requireDefined } from "./assertions";
+import { initMultiSourceHttpClientState, MultiSourceHttpClient, aggregateArrays } from "./Http";
+import { Signer, SignCallback } from "./Signer";
+
+export interface AddedOn {
+    addedOn: string;
+}
+
+export interface Published {
+    published: boolean;
+}
+
+/**
+ * Blockchain File, extended with private attributes and timestamp.
+ */
+export interface LocFile extends BlockchainFile, Partial<AddedOn> {
+    name: string;
+}
+
+export interface MergedFile extends LocFile, Published {
+}
+
+/**
+ * Blockchain MetadataItem, extended with timestamp.
+ */
+export interface LocMetadataItem extends MetadataItem, Partial<AddedOn> {
+}
+
+export interface MergedMetadataItem extends LocMetadataItem, Published {
+}
+
+/**
+ * Blockchain MetadataItem, extended with timestamp.
+ */
+export interface LocLink extends Link, AddedOn {
+    target: string; // is redundant with inherited "id: UUID"
+}
+
+export interface MergedLink extends LocLink, Published {
+}
+
+export interface LocCollectionItem extends AddedOn {
+    itemId: string,
+}
+
+export type MergedCollectionItem = CollectionItem & Partial<AddedOn>
+
+export interface LocRequestVoidInfo {
+    reason?: string; // undefined in public view
+    voidedOn?: string;
+}
+
+export interface LocRequest {
+    ownerAddress: string;
+    requesterAddress?: string | null;
+    requesterIdentityLoc?: string | null;
+    description: string;
+    locType: LocType;
+    createdOn: string;
+    decisionOn?: string;
+    id: string;
+    status: LocRequestStatus;
+    rejectReason?: string;
+    userIdentity?: UserIdentity;
+    closedOn?: string;
+    files: LocFile[];
+    metadata: LocMetadataItem[];
+    links: LocLink[];
+}
+
+export interface LocData {
```

Comment:
I would move this definition to `Loc.ts` as `LocData` objects are never instantiated in this module.

Reply:
Done.